upgrade/scripts/upgrade/util/upgrade-cray-nls.sh

```diff
@@ -50,38 +50,57 @@ function patchKeycloak() {
     --data-urlencode username="$(kubectl get secrets keycloak-master-admin-auth -o jsonpath='{.data.user}' -n services | base64 -d)" \
     --data-urlencode password="$(kubectl get secrets keycloak-master-admin-auth -o jsonpath='{.data.password}' -n services | base64 -d)" \
     https://api-gw-service-nmn.local/keycloak/realms/master/protocol/openid-connect/token | jq -r '.access_token')
-    #shellcheck disable=SC2126
-    foundArgoInRedirectUris=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq '.[] | select(.clientId=="oauth2-proxy-customer-management") | .redirectUris' | grep "https://argo.cmn" | wc -l)
-    redirectUris=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .redirectUris')
-    #shellcheck disable=SC2126
-    foundArgoInWebOrigins=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq '.[] | select(.clientId=="oauth2-proxy-customer-management") | .webOrigins' | grep "https://argo.cmn" | wc -l)
-    webOrigins=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .webOrigins')
 
-    id=$(curl -s -k -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients" | jq -r '.[] | select(.clientId=="oauth2-proxy-customer-management") | .id')
+    res_file=$(mktemp)
+    http_code=$(curl -s -o "${res_file}" -w "%{http_code}" -k -XGET -H "Authorization: Bearer ${TOKEN}" "https://api-gw-service-nmn.local/keycloak/admin/realms/shasta/clients")
+    if [[ ${http_code} -ne 200 ]]; then
```

Comment:
We aren't going to hit this if the curl fails because of the set -e, right?   That will cause it to exit right after line 55.

Reply:
Did you test any negative tests where it hits the error conditions?

Reply:
It will if the curl succeeds and returns something like 404 or 500.  But it will not work, for example, if DNS is not working and it cannot resolve api-gw-service-nmn.local.

Reply:
One other concern that I have is that it looks like prerequisites.sh will not fail when upgrade-cray-nls.sh fails because it is in a subshell.   I think it will just move on to the next stage and we won't notice that something went wrong.   Am I reading that wrong?

Reply:
you are right ...... i might just throw out shell really... why can't it give me an error object like a real language would .... i need to babysit every tiny bits. I hate shell/bash more and more everyday

Reply:
I think if you just check ahead at the beginning to make sure the TOKEN is valid, that should cover the case where it cannot reach the gateway because of networking or DNS.

Reply:
throw out bash, problem solved